sphinx_gallery/gen_rst.py

```diff
@@ -231,6 +231,19 @@ def text2string(content):
         return content
 
 
+def extract_thumbnail_number(text):
+    """ Pull out the thumbnail image number specified in the docstring. """
+
+    # check whether the user has specified a specific thumbnail image
+    pattr = re.compile("^\s*#\s*sphinx_gallery_thumbnail_number\s*=\s*([0-9]+)\s*$", flags=re.MULTILINE)
+    match = pattr.search(text)
+    try:
+        thumbnail_number = int(match.groups()[0])
```

Comment:
I would do:
```python
thumbnail_number = 1 if match is None else int(match.groups()[0])
```

I am not convinced asking for forgiveness rather than permission is better in this particular case.